File: src/utils/units.ts
export type WeightUnit = 'kg' | 'lb';

const CONVERSION_FACTOR = 2.20462;

function roundTwoDecimals(value: number): number {
  return Math.round(value * 100) / 100;
}

export function kgToLb(kg: number): number {
  return roundTwoDecimals(kg * CONVERSION_FACTOR);
}

export function lbToKg(lb: number): number {
  return roundTwoDecimals(lb / CONVERSION_FACTOR);
}

export function convertWeight(value: number, from: WeightUnit, to: WeightUnit): number {
  if (from === to) return value;
  return from === 'kg' ? kgToLb(value) : lbToKg(value);
}

export function formatWeight(value: number, unit: WeightUnit): string {
  const rounded = roundTwoDecimals(value);
  return `${rounded} ${unit}`;
}

export function formatWeightShort(value: number, _unit: WeightUnit): string {
  const rounded = roundTwoDecimals(value);
  return `${rounded}`;
}

export function getIncrement(unit: WeightUnit): number {
  return unit === 'kg' ? 2.5 : 5;
}

File: flutter-version/lib/features/settings/domain/units.dart
enum WeightUnit {
  kg('kg'),
  lb('lb');

  const WeightUnit(this.value);

  final String value;

  static WeightUnit fromText(String value) {
    return switch (value) {
      'lb' => WeightUnit.lb,
      _ => WeightUnit.kg,
    };
  }
}

const _kgToLbFactor = 2.20462;

double kgToLb(double kg) {
  return _roundTwoDecimals(kg * _kgToLbFactor);
}

double lbToKg(double lb) {
  return _roundTwoDecimals(lb / _kgToLbFactor);
}

double convertWeight(
  double value, {
  required WeightUnit from,
  required WeightUnit to,
}) {
  if (from == to) {
    return value;
  }

  return from == WeightUnit.kg ? kgToLb(value) : lbToKg(value);
}

String formatWeight(double value, WeightUnit unit) {
  return '${formatWeightShort(value)} ${unit.value}';
}

String formatWeightShort(double value) {
  final rounded = _roundTwoDecimals(value);
  if (rounded % 1 == 0) {
    return rounded.toInt().toString();
  }

  return rounded.toString();
}

double getIncrement(WeightUnit unit) {
  return unit == WeightUnit.kg ? 2.5 : 5;
}

double _roundTwoDecimals(double value) {
  return (value * 100).roundToDouble() / 100;
}
